### DISCUSSION TITLE
`keycloak-admin-client-reactive` what is the differnce to `keycloak-admin-client`

### DISCUSSION THREAD

**Discussion by agreedSkiing** [quarkusio/quarkus#38951] [Q&A] [answered]
Started to wonder what is the difference between `keycloak-admin-client-reactive` and `keycloak-admin-client`?

What we, the team I work in, have noticed is that we still need to do a runSubscriptionOn for our `keycloak-admin-client-reactive` to work well with the rest of our reactiv code.
~~~java
return Uni
    .createFrom()
    .item(() -> keycloak.realm(realm).users().count())
    .invoke(() -> log.infov("Check that realm client {0} is up and running", clientId))
    .runSubscriptionOn(Infrastructure.getDefaultWorkerPool())
~~~

P.S. This is also a continuation from https://github.com/quarkusio/quarkus/issues/38874#issuecomment-1953915062 question after the issue was closed.

[a comment by an automated account is not shown]

**Reply by sberyozkin** [reply, marked as answer]
@agreedSkiing As far as I recall the primary reason for introducing the `-reactive` extension was to make sure it could be used alongside RestEasy Reactive because otherwise keycloak-admin-client depends on the classic RestEasy and it was causing conflicts.
You are right though that other than that, these extensions are no different to each other. However, once the Keycloak team introduces a Uni-aware version of its admin API, then we'll be able to differentiate.
@geoand Please correct me if I've missed something

**Reply by geoand** [reply]
Correct
